src/plugins/event-plugins/DomEventPlugin.ts

```diff
@@ -36,13 +41,18 @@ const defaultConfig: DomEventPluginConfig = {
 export class DomEventPlugin implements Plugin {
     private recordEvent: RecordEvent | undefined;
     private pluginId: string;
-    private eventListenerMap: Map<TargetDomEvent, EventListener>;
+    private eventListenerMap: Map<TargetDomEvent, Map<string, EventListener>>;
```

Comment:
**suggestion:** The problem we have here is that the TargetDomEvent can potentially contain three selectors: ID, CSS locator or object. Having a map within a map makes things a little hard to follow. Consider a different strategy:

1. Map a `TargetDomEvent` to an array of `EventListener`. There will be either one or two items in the array. This will remove a lot of the complexity, because we don't need to think about the 'type' of event listener.

2. Only use one of the three selectors. This adds the least complexity. We can document that only one of selectors should be given, and the order in which the selectors will be preferred.

Reply:
Will go with option 2. Preference will be: CSS Selector > Element ID > Element. CSS Selector will be preferred if provided because it point to the most elements while element ID points to only one element.